torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1009,9 +1009,28 @@ void ProcessGroupNCCL::broadcastUniqueNCCLID(
         reinterpret_cast<uint8_t*>(ncclID) + NCCL_UNIQUE_ID_BYTES);
     store_->set(storeKey, vec);
   } else {
-    auto vec = store_->get(storeKey);
-    TORCH_CHECK(vec.size() == NCCL_UNIQUE_ID_BYTES);
-    std::memcpy(ncclID, vec.data(), vec.size());
+    std::string exceptionMsg = c10::str(
+        "[",
+        rank_,
+        "] is setting up NCCL communicator and "
+        "retreiving ncclUniqueId from [0] via c10d key-value store by key '",
+        storeKey,
+        "', but store->get('",
+        storeKey,
+        "') got error: ");
+    try {
+      auto vec = store_->get(storeKey);
+      TORCH_CHECK(vec.size() == NCCL_UNIQUE_ID_BYTES);
+      std::memcpy(ncclID, vec.data(), vec.size());
+    } catch (const std::runtime_error& e) {
```

Comment:
Can we just use 
```
catch (const std::exception& e) {
    TORCH_CHECK(false, exceptionMsg + e.what())
}
catch (...) {
    TORCH_CHECK(false, "Unknown exception")
}
```

the reason is that TORCH_CHECK works well with TORCH_CPP_SHOW_STACKTRACES so we generally prefer throwing those exceptions IIUC.

Reply:
Updated to TORCH_CHECK()